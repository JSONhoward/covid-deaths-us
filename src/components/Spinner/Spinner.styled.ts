import styled, {keyframes} from 'styled-components'

const rotateSpinner = keyframes`
0% {
    transform: rotate(0deg);
}
100% {
    transform: rotate(360deg);
}
`

export const SpinnerContainer = styled('div')`
position: relative;
display: grid;
place-items: center;
height: 100%;
min-height: calc(100vh - 8rem);
width: 100%;

img {
    height: 5rem;
    width: auto;
    animation: 2s ${rotateSpinner} linear;
}
`