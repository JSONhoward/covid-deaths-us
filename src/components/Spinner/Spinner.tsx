import React from 'react'

import CovidSpinner from '../../images/covid-image.png'
import { SpinnerContainer } from './Spinner.styled';

const Spinner = () => {
    return (
        <SpinnerContainer>
            <img data-testid='loadingSpinnerImage' src={CovidSpinner} alt={'Covid-19 Virus'} />
        </SpinnerContainer>
    )
}

export default Spinner
