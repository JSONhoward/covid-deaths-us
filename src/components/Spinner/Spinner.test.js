import React from 'react'
import {create} from 'react-test-renderer'
import {screen, render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Spinner from './'
import CovidSpinner from '../../images/covid-image.png'

describe('Spinner Tests',() => {
    test('Snapshot Test', () => {
        const tree = create(<Spinner />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Spinner Image is Rendered', () => {
        render(<Spinner />)

        expect(screen.getByTestId('loadingSpinnerImage')).toHaveAttribute('src', CovidSpinner)
    })
})