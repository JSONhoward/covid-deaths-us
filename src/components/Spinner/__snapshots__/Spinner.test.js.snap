// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Tests Snapshot Test 1`] = `
<div
  className="sc-bdnylx bZWmCJ"
>
  <img
    alt="Covid-19 Virus"
    data-testid="loadingSpinnerImage"
    src="covid-image.png"
  />
</div>
`;
